weave-js/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,75 @@
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+import React from 'react';
+import {twMerge} from 'tailwind-merge';
+
+import {Button} from '../Button';
+import {IconName} from '../Icon';
+import {Tailwind} from '../Tailwind';
+import {ToggleButtonGroupSizes} from './types';
+
+export type ToggleOption = {
+  label: string;
+  icon?: IconName;
+};
+
+export type ToggleButtonGroupProps = {
+  options: ToggleOption[];
+  value: string;
+  size: ToggleButtonGroupSizes;
+  isDisabled?: boolean;
+  onValueChange: (value: string) => void;
+};
+
+/**
+ * ToggleButtonGroup component should only be rendered if options.length >= 2.
+ */
+export const ToggleButtonGroup = React.forwardRef<
+  HTMLDivElement,
+  ToggleButtonGroupProps
+>(({options, value, size, isDisabled = false, onValueChange}, ref) => {
+  if (options.length < 2) {
+    return null; // Do not render if there are fewer than two options
+  }
+
```

Comment:
Could consider logging a warning if the provided value is not one of the options.